freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/verticalscale/actions/FreeIPAVerticalScaleActions.java

```diff
@@ -0,0 +1,183 @@
+                            .withCrn(stack.getResourceCrn())
+                            .withPlatform(stack.getCloudPlatform())
+                            .withVariant(stack.getPlatformvariant())
+                            .withLocation(location)
+                            .withAccountId(Crn.safeFromString(stack.getResourceCrn()).getAccountId())
+                            .build();
+
+                    FreeIPAVerticalScaleRequest request = new FreeIPAVerticalScaleRequest(
+                            cloudContext,
+                            cloudCredential,
+                            cloudStack,
+                            cloudResources,
+                            freeIPAVerticalScaleRequest);
+
+                    sendEvent(ctx, request);
+                } catch (Exception e) {
+                    LOGGER.error("Failed to Vertical scaling FreeIPA", e);
+                    sendEvent(ctx, STACK_VERTICALSCALE_FINISHED_FAILURE_EVENT.selector(),
+                            new UpscaleFailureEvent(stack.getId(), "Vertical Scale update", Set.of(), Map.of(), e));
+                }
+            }
+        };
+    }
+
+    @Bean(name = "STACK_VERTICALSCALE_FINISHED_STATE")
+    public Action<?, ?> verticalScaleFinished() {
+        return new AbstractFreeIPAVerticalScaleAction<>(FreeIPAVerticalScaleResult.class) {
+
+            @Inject
+            private OperationService operationService;
+
+            @Override
+            protected void doExecute(StackContext context, FreeIPAVerticalScaleResult payload, Map<Object, Object> variables) {
+                freeIPAVerticalScaleService.updateTemplateWithVerticalScaleInformation(context.getStack().getId(), payload.getFreeIPAVerticalScaleV1Request());
+                stackUpdater.updateStackStatus(context.getStack().getId(), DetailedStackStatus.STOPPED, "Vertical scale complete");
+                sendEvent(context, STACK_VERTICALSCALE_FINALIZED_EVENT.selector(), payload);
+            }
+        };
+    }
+
+    @Bean(name = "STACK_VERTICALSCALE_FAILED_STATE")
+    public Action<?, ?> verticalScaleFailedAction() {
+        return new AbstractFreeIPAVerticalScaleAction<>(FreeIPAVerticalScaleFailureEvent.class) {
+            @Inject
+            private OperationService operationService;
+
+            @Override
+            protected StackContext createFlowContext(FlowParameters flowParameters, StateContext<FreeIPAVerticalScaleState,
+                    FreeIPAVerticalScaleEvent> stateContext,
+                    FreeIPAVerticalScaleFailureEvent payload) {
+                Flow flow = getFlow(flowParameters.getFlowId());
+                flow.setFlowFailed(payload.getException());
+                return super.createFlowContext(flowParameters, stateContext, payload);
+            }
+
+            @Override
+            protected void doExecute(StackContext context, FreeIPAVerticalScaleFailureEvent payload, Map<Object, Object> variables) {
```

Comment:
could you add a log somewhere, as this is the failure action, we should have logs